From: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
Date: Tue, 15 Sep 2026 06:59:20 +0000
Subject: [PATCH] drm: Skip empty dropped commit cleanup

Queue main thread cleanup only when the commit thread retained objects
for deletion. Normal pageflips should not cause an unnecessary event loop
wakeup.


(cherry picked from commit a8ccec655261f502610c569c03b7c839436c4238)

Co-authored-by: Erhan Bilgili <erhan.bilgili@gmail.com>

Upstream-Status: Backport [https://invent.kde.org/plasma/kwin/-/commit/2856eecbfd4174793aba230573d6293435652d8e]
---
--- a/src/backends/drm/drm_commit_thread.cpp
+++ b/src/backends/drm/drm_commit_thread.cpp
@@ -198,7 +198,9 @@
         m_commits.clear();
         qCWarning(KWIN_DRM) << "atomic commit failed:" << strerror(errno);
     }
-    QMetaObject::invokeMethod(this, &DrmCommitThread::clearDroppedCommits, Qt::ConnectionType::QueuedConnection);
+    if (!m_commitsToDelete.empty()) {
+        QMetaObject::invokeMethod(this, &DrmCommitThread::clearDroppedCommits, Qt::ConnectionType::QueuedConnection);
+    }
 }
 
 void DrmCommitThread::optimizeCommits(TimePoint pageflipTarget)
